### ISSUE TITLE
Problems while running train_tracker.sh

### ISSUE THREAD

**Issue by SMZCC** [nrupatunga/PY-GOTURN#5]
#### When I run the file of  train_tracker.sh,I got a problem :
```
[libprotobuf ERROR google/protobuf/text_format.cc:298] Error parsing text-format  caffe.SolverParameter: 12:14: Expected integer, got: RANDOM_SEED
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1009 21:19:19.955904  3587 upgrade_proto.cpp:1101] Check failed:  ReadProtoFromTextFile(param_file, param) Failed to parse SolverParameter file: ./nets/solver.prototxt
 *** Check failure stack trace: ***
train_tracker.sh: 行 16:  3587 已放弃               (核心已转储) python -m goturn.train.train --imagenet $IMAGENET_FOLDER --alov $ALOV_FOLDER --init_caffemodel $INIT_CAFFEMODEL --train_prototxt $TRACKER_PROTO --solver_prototxt $SOLVER_PROTO --lamda_shift 5 --lamda_scale 15 --min_scale -0.4 --max_scale 0.4 --gpu_id 0
```
#### Thanks

**Comment by nrupatunga** [comment]
In the [solver.prototxt](https://github.com/nrupatunga/PY-GOTURN/blob/880870b093a4376231b568b1b9188bd525adc30d/nets/solver.prototxt) remove 
this line 
`random_seed: RANDOM_SEED`

Note: Not tested from my end. Please do check.

**Comment by SMZCC** [comment]
- I can remove that line without any effect ?
- What's the function of that line ? @nrupatunga

**Comment by SMZCC** [comment]
- Can I  remove that line without any effect ?
- What's the function of that line ? @nrupatunga

**Comment by SMZCC** [comment]
- Can I remove that line without any effection to  the whole program?
- What's the function of that line ? Or it is just a redundancy？ @nrupatunga

**Comment by nrupatunga** [comment]
Random seed is set [here](https://github.com/nrupatunga/PY-GOTURN/blob/9bdf7b0d13faa814bf4efa699811d4b1082dac6f/goturn/train/train.py#L75). So you can remove the line

**Comment by SMZCC** [comment]
#### Thanks I have got that @nrupatunga

**Comment by SMZCC** [comment]
#### Dear nrupatunga, can I do the same with DEVICE_ID ,because the similar situation has occurred with DEVICE_ID ?However, I didn't find the DEVICE_ID other than in solver.prototxt
```
[libprotobuf ERROR google/protobuf/text_format.cc:298] Error parsing text-format caffe.SolverParameter: 12:12: Expected integer, got: DEVICE_ID
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1010 14:50:55.883632  2807 upgrade_proto.cpp:1101] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse SolverParameter file: ./nets/solver.prototxt
*** Check failure stack trace: ***
train_tracker.sh: 行 16:  2807 已放弃               (核心已转储) python -m goturn.train.train --imagenet $IMAGENET_FOLDER --alov $ALOV_FOLDER --init_caffemodel $INIT_CAFFEMODEL --train_prototxt $TRACKER_PROTO --solver_prototxt $SOLVER_PROTO --lamda_shift 5 --lamda_scale 15 --min_scale -0.4 --max_scale 0.4 --gpu_id 0

```

**Comment by SMZCC** [comment]
#### Dear nrupatunga, in the file of solver.prototxt , I find the parameter: net  which confuses me in some way:
- the parameter, net, can't be removed, I have tried, then how to set it's value while we already have a init_model?
- Why did we need a init_model? 
- Can't we train a model from scratch? Then how to do that?
#### Thanks  @nrupatunga

**Comment by nrupatunga** [comment]
I will clean up `solver.prototxt`, which will help you understand better. Thanks

**Comment by SMZCC** [comment]
- Thank you very much
- However, I'm now very curious about last three questions .I eager a key  :sob: 
@nrupatunga

**Comment by nrupatunga** [comment]
@SMZCC 
The issue with `solver.prototxt` is fixed. Please do check the updated code. 

> the parameter, net, can't be removed, I have tried, then how to set it's value while we already have a init_model?
- This issue is fixed in updated code

> Why did we need a init_model?

- This is out of scope of the discussion. For further understanding please refer [here] (http://cs231n.github.io/transfer-learning/)

> Can't we train a model from scratch? Then how to do that?

- You can. Removing this [line](https://github.com/nrupatunga/PY-GOTURN/blob/4f77ce6a1f41d47ece48931fc191e6a235071b17/goturn/network/regressor.py#L121) should work

**Comment by SMZCC** [comment]
#### Thank you very much . I've got that.
#### You are so strong that my way is still long.

**Comment by nrupatunga** [comment]
Closing this issue. Feel free to reopen if you find an issue

**Comment by SMZCC** [comment]
#### Dear nrupatunga 
Hello
I have some questions about the code while running `train_tracker.sh` as followings:
- the parameter , max_iter ,in the file of solver_prototxt , doesn't seem to work  when I modified the value of it to 1000, while it was still on training  to inter_num 1001, 1002....,it didn't stop at iter_num of 1000. Why?

- I can't understand how the caffe receive its training data and bbox. I searched for it , then I find most of the net_prototxt is look like this:
```python
layer {
  name: "cifar"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mean_file: "examples/cifar10/mean.binaryproto"
  }
  data_param {
    source: "examples/cifar10/cifar10_train_lmdb"   # here it shows the training data
    batch_size: 100
    backend: LMDB
  }
}
``` 
while the prototxt we used is: 
```python
name: "CaffeNet"

input: "target"
input: "image"
input: "bbox"

#target
input_dim: 1
input_dim: 3
input_dim: 227
input_dim: 227

#image
input_dim: 1
input_dim: 3
input_dim: 227
input_dim: 227

#bbox
input_dim: 1
input_dim: 4
input_dim: 1
input_dim: 1

layer {
  name: "conv1"
  type: "Convolution"
  bottom: "target"
  top: "conv1"
  param {
    lr_mult: 0
    decay_mult: 1
  }
  param {
    lr_mult: 0
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
```
I don't know which indicates the data the caffe will receive. Why will the caffe get the training data with writing like this ? 

- I should can't find the formula of loss which confuses me a lot .If I want to modify the loss formula ,how should I do ?
- The loss is always up and down around 150 . Is that normal?Doesn't it seem too large?
- How could I see the predicted bbox and the gt  during training phase  ?
- Is the net structure defined in tracker_prototxt the same with the caffe_model_init given ? 

#### Thanks a lot 
@nrupatunga
